t/api/03-auth.t

```diff
@@ -42,6 +43,11 @@ my $app = $t->app;
 $t->ua(OpenQA::Client->new()->ioloop(Mojo::IOLoop->singleton));
 $t->app($app);
 
+# we don't want to *actually* delete any assets when we're testing
+# whether we're allowed to or not, so let's mock that out
+my $mock_asset = Test::MockModule->new('OpenQA::Schema::Result::Assets');
+$mock_asset->mock(remove_from_disk => sub { return 1; });
```

Comment:
It should also be tested that remove_from_disk is actually called here.

Reply:
I don't think we need to. The `02-assets.t` test already tests deletion, using ISOs it creates and removes and thus which don't interfere with any other tests. This test is not really a test of the deletion functionality, it is a test that authorization is working properly.

Reply:
No, i mean you should test that the mock actually does something. So when things are refactored it gets noticed this is no longer needed.

Reply:
oh, or do you mean just as a canary for if the function gets moved or renamed or something? if so, yeah, I guess that makes sense.